tuplex/core/src/physical/TransformTask.cc

```diff
@@ -751,31 +753,43 @@ namespace tuplex {
 
         // completely compiled parser or the smaller version?
         if(cellBasedFunctor) {
-            if(fmt == FileFormat::OUTFMT_CSV) {
-                auto csv = new CSVReader(this, reinterpret_cast<codegen::cells_row_f>(_functor), makeParseExceptionsInternal, operatorID, exceptionCallback(), numColumns, delimiter,
-                                         quotechar, colsToKeep);
-                csv->setRange(rangeStart, rangeStart + rangeSize);
-                csv->setHeader(header);
-                _reader.reset(csv);
-            } else if(fmt == FileFormat::OUTFMT_TEXT) {
-                auto text = new TextReader(this, reinterpret_cast<codegen::cells_row_f>(_functor));
-                text->setRange(rangeStart, rangeStart+rangeSize);
-                _reader.reset(text);
-            } else {
-                throw std::runtime_error("unsupported input filetype");
+            switch (fmt) {
+                case FileFormat::OUTFMT_CSV: {
+                    auto csv = new CSVReader(this, reinterpret_cast<codegen::cells_row_f>(_functor), makeParseExceptionsInternal, operatorID, exceptionCallback(), numColumns, delimiter,
+                                             quotechar, colsToKeep);
+                    csv->setRange(rangeStart, rangeStart + rangeSize);
+                    csv->setHeader(header);
+                    _reader.reset(csv);
+                    break;
+                }
+                case FileFormat::OUTFMT_TEXT: {
```

Comment:
in the above functions: Do we support OUTFMT_TEXT? We should add extensions etc.